### PR TITLE
Preload assets in Rollup apps - fixes #408

### LINKED ISSUE #408: Preloading assets in Rollup apps
Currently, only `client.xyz123.js` is preloaded. Sapper should add preload headers for all the JS and CSS needed for the page being server-rendered

### PR DESCRIPTION
**Pull request by nsivertsen**
I'm not sure if the way I'm testing whether a chunk is a page is 100% right, see `RollupResult.ts`, lines 47-48.

Fixes #408

### CODE REVIEW

File: src/core/create_compilers/RollupResult.ts [hunk 1]
```diff
@@ -33,14 +35,21 @@ export default class RollupResult implements CompileResult {
...
 
+		const routes_dir = locations.routes();
+
 		compiler.chunks.forEach(chunk => {
 			if (compiler.input in chunk.modules) {
 				this.assets.main = chunk.fileName;
+			} else {
+				Object.keys(chunk.modules).forEach((module: string) => {
+					if (!module.startsWith(routes_dir)) return;
+					if (/\.css$/.test(module)) return;
```
Comment (nsivertsen) [inline src/core/create_compilers/RollupResult.ts]: Not sure whether this is 100% right...

**Comment by benmccann** [comment]
@nsivertsen is this something you'd still be interested in pursuing? I know it's been awhile, but any chance you'd be able to rebase it?

**Comment by nsivertsen** [comment]
@benmccann hi! I would be, however I'm quite swamped at the moment so it would be a couple weeks before I could look into it. If you have time sooner, feel free to take a stab at it!

As both Sapper and Rollup have gone through a fair amount of changes in the meantime, there might be a better way of doing this, so it could be worth looking at this with fresh eyes.

**Comment by benmccann** [comment]
No worries. We can wait for whenever you get a chance

**Comment by benmccann** [comment]
I've sent https://github.com/sveltejs/sapper/pull/1269 which is rebased against master and handles all transitive JS and CSS files

**Comment by antony** [comment]
Closing in favour of https://github.com/sveltejs/sapper/pull/1269

**Pull request closed without merging**